src/neo/SmartContract/Native/ContractManagement.cs

```diff
@@ -220,13 +223,7 @@ private void Update(ApplicationEngine engine, byte[] nefFile, byte[] manifest, S
             }
             Check(contract.Nef.Script, contract.Manifest.Abi);
             contract.UpdateCounter++; // Increase update counter
-            if (nefFile != null)
-            {
-                ContractMethodDescriptor md = contract.Manifest.Abi.GetMethod("_deploy", 2);
-                if (md != null)
-                    engine.CallFromNativeContract(Hash, contract.Hash, md.Name, data, true);
-            }
-            engine.SendNotification(Hash, "Update", new VM.Types.Array { contract.Hash.ToArray() });
+            return OnDeploy(engine, contract, data, true);
```

Comment:
Require await?

Reply:
It's not an `async` method. So we only need to return the `ContractTask`.